src/Query.tsx

```diff
@@ -70,7 +70,7 @@ export interface QueryState<TData = any> {
 }
 
 class Query<TData = any> extends React.Component<
-  QueryProps,
+  QueryProps<TData>,
```

Comment:
Can you also update the type definition of the props in the constructor.

Reply:
Good catch